features/invite/impl/src/main/kotlin/io/element/android/features/invite/impl/response/AcceptDeclineInvitePresenter.kt

```diff
@@ -35,8 +35,13 @@ class AcceptDeclineInvitePresenter @Inject constructor(
     private val client: MatrixClient,
     private val joinRoom: JoinRoom,
     private val notificationCleaner: NotificationCleaner,
-    private val seenInvitesStore: SeenInvitesStore,
+    seenInvitesStoreFactory: SeenInvitesStoreFactory,
 ) : Presenter<AcceptDeclineInviteState> {
+    private val seenInvitesStore = seenInvitesStoreFactory.getOrCreate(
+        sessionId = client.sessionId,
+        sessionCoroutineScope = client.sessionCoroutineScope,
+    )
```

Comment:
I don't mind using this, but I think you could also create a `@Provides` method for `SessionScope` that would return the cached store if needed:

```kotlin
@Provides
fun providesSeenInvitesStore(factory: SeenInvitesStoreFactory, sessionId: SessionId, @SessionCoroutineScope sessionCoroutineScope: CoroutineScope) {
    return factory.getOrCreate(sessionId, sesionCoroutineScope)
 }
 ```
 
 And then you can just inject the result in the session scope.

Reply:
Ah that's a bit cleaner, will give it a try

Reply:
80156681aeafdd0b4457ed894c046d94b492e194